desktop/main/NodeConfig.ts

```diff
@@ -21,33 +28,188 @@ export const loadNodeConfig = async (): Promise<NodeConfig> =>
+        filePath,
       },
       error
     );
   }
+};
+
+export const saveSmeshingOptsInCustomConfig = async (
+  netName: string,
+  opts: Partial<NodeConfig['smeshing']>
+): Promise<Partial<NodeConfig>> => {
+  const customConfig = await loadCustomNodeConfig(netName);
+
+  customConfig.smeshing = opts;
+
+  await writeCustomNodeConfig(netName, customConfig);
+
+  return customConfig;
+};
+
+export const detectStoreServiceMigrationForSmeshingOpts = (
+  opts: Partial<NodeConfig['smeshing']> | undefined,
+  genesisID: string
+): Partial<NodeConfig['smeshing']> =>
+  opts || safeSmeshingOpts(undefined, genesisID); // set default dir path
+
+const createCustomNodeConfig = async (
+  netName: string,
+  genesisID: string
+): Promise<Partial<NodeConfig>> => {
+  const opts: Partial<NodeConfig['smeshing']> | undefined = StoreService.get(
+    `smeshing.${genesisID}`
+  );
+
+  // migrate options from StoreService or place only default opts
+  const smeshingOpts = detectStoreServiceMigrationForSmeshingOpts(
+    opts,
+    genesisID
+  );
+
+  const config = await saveSmeshingOptsInCustomConfig(netName, smeshingOpts);
+
+  StoreService.remove(`smeshing.${genesisID}`);
+
+  return config;
+};
+export const loadOrCreateCustomConfig = async (
+  netName: string,
+  genesisID: string
+): Promise<Partial<NodeConfig>> =>
+  existsSync(path.join(USERDATA_DIR, getCustomNodeConfigName(netName)))
+    ? loadCustomNodeConfig(netName)
+    : createCustomNodeConfig(netName, genesisID);
+
+export const getNodeConfigByMerge = (
+  config: Partial<NodeConfig>,
+  customConfig: Partial<NodeConfig>
+): NodeConfig => R.mergeLeft(customConfig, config);
+
+export const getMergedSmeshingOpts = (
+  config: Partial<NodeConfig>,
+  update: Partial<NodeConfig['smeshing']>,
+  genesisID: string
+) => {
+  const defaultSmeshingOpts = R.isEmpty(update)
+    ? safeSmeshingOpts(undefined, genesisID)
+    : {};
+  const migrateSmeshingOpts = R.isEmpty(update) ? {} : config.smeshing;
+
+  const smeshingOpts = {
+    ...migrateSmeshingOpts,
+    ...defaultSmeshingOpts,
+    ...update,
+  };
+
+  const batchSize =
+    smeshingOpts?.['smeshing-opts']?.['smeshing-opts-compute-batch-size'];
+  if (
+    batchSize &&
+    batchSize === DEFAULT_SMESHING_BATCH_SIZE &&
+    smeshingOpts['smeshing-opts']
+  ) {
+    smeshingOpts['smeshing-opts'][
+      'smeshing-opts-compute-batch-size'
+    ] = R.pathOr(
+      DEFAULT_SMESHING_BATCH_SIZE,
+      ['smeshing-opts', 'smeshing-opts-compute-batch-size'],
+      config?.smeshing
+    );
+  }
+
+  return smeshingOpts;
+};
+
+export const updateSmeshingOpts = async (
+  netName: string,
+  updateSmeshingOpts: Partial<NodeConfig['smeshing']>
+): Promise<Partial<NodeConfig>> => {
+  const customNodeConfig = await loadCustomNodeConfig(netName);
+  const clientConfig = await loadNodeConfig();
+
+  const smeshingOpts = getMergedSmeshingOpts(
+    customNodeConfig,
+    updateSmeshingOpts,
+    generateGenesisIDFromConfig(clientConfig)
+  );
```

Comment:
```suggestion
  const smeshingOpts = safeSmeshingOpts(
    {
      ...(customNodeConfig.smeshing || {}),
      ...updateSmeshingOpts,
    },
    generateGenesisIDFromConfig(clientConfig)
  );
```

And please, rename the `updateSmeshingOpts` argument. It should not (1) sound like a function, (2) collide with the function name.

Reply:
on [delete](https://github.com/spacemeshos/smapp/pull/1248/files#diff-d5d2c5e79a5b1dd012409202c6df9fb57267b51e876e8835631a1520b7c4a9afR413), we need to ignore smeshing opts from custom config ,I moved on top [the code](https://github.com/spacemeshos/smapp/pull/1248/files#diff-76a76c53daab14f3dc9b8262657b113cc63889c0cbc8ac953776d3f9cd09e3e5R129), but still not so simple.